rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+which uses an account-based model. Instead, the private state employs a UTXO (Unspent Transaction Output)-based model. In this model, each transaction output is a discrete 
+unit that can be independently spent in future transactions. This design provides users with granular control over their transaction outputs.
+
+A key aspect of maintaining privacy within the private state is the use of Zero-Knowledge Proofs (ZKPs). ZKPs allow transactions to be validated without revealing any 
+underlying private data. This means that while the system can verify that a transaction is legitimate, the details of the transaction remain confidential. Only parties 
+with the appropriate viewing key can access and reconstruct the user’s list of UTXOs, ensuring that sensitive information is protected.
+
+The private state also employs a dual-storage approach to balance privacy and transparency. UTXOs are stored in plaintext within Sparse Merkle Trees (SMTs) in the private state, 
+providing detailed and accessible records for the user. In contrast, the public state only holds hashes of these UTXOs. This method ensures that while the public can verify 
+the existence and integrity of private transactions through these hashes, they cannot access the specific details.
+
+The workflow for a transaction in the private state begins with the user initiating a transaction involving their private data, such as transferring a private balance or 
+updating private smart contract storage. The transaction involves spending existing UTXOs, represented as leaves in the Merkle tree, and creating new UTXOs, 
+which are then appended to the user’s private list. The zkVM (Zero-Knowledge Virtual Machine) generates a zero-knowledge proof to validate the transaction without revealing 
+any private data, ensuring the transaction adheres to the system's rules.
+
+Once the proof is generated, it is submitted to the sequencer, which verifies the transaction’s validity. Upon successful verification, the nullifier is added to the nullifier set, 
+preventing double-spending of the same UTXO. The use of ZKPs and nullifiers ensures that the private state maintains both security and privacy.
+
+In summary, the private state in NSSA is meticulously designed to provide users with control over their private information while ensuring the security and integrity of transactions. 
+By utilizing a UTXO-based model, individual Merkle trees, Zero-Knowledge Proofs, and a dual-storage system, NSSA achieves a balance between confidentiality and verifiability, 
+making it a robust solution for managing private blockchain transactions.
+
+### c) zkVM (Zero-Knowledge Virtual Machine)
+---
+
+The Zero-Knowledge Virtual Machine (zkVM) is a pivotal component in the Nescience State Separation Architecture (NSSA), designed to uphold the highest standards 
```

Comment:
Replace:
"The Zero-Knowledge Virtual Machine (zkVM)"

With:
"The zkVM"
----
Replace:
"the Nescience State Separation Architecture (NSSA)"

With:
"NSSA"